### ISSUE TITLE
Type hints need to include `requests_mock.ANY`

### ISSUE THREAD

**Issue by lmmarsano** [jamielennox/requests-mock#171]
According to [request matching documentation](https://requests-mock.readthedocs.io/en/latest/matching.html#matching-any),

https://github.com/jamielennox/requests-mock/blob/850671f958afd287c2dbce7420f670215892314f/requests_mock/mocker.pyi#L33

should be

```python
method: Union[str, typing.Literal[requests_mock.adapter.ANY]]
```

etc.

**Comment by lmmarsano** [comment]
> Unfortunately `typing.Literal[requests_mock.adapter.ANY]` is an invalid type; `Literal` only supports `None`, bool, bytes, int, str, or an enum value.

Good point.
Maybe that's a good argument for the next version making `requests_mock.adapter.ANY` an enum.
@jamielennox, would that be okay?

**Comment by jamielennox** [comment]
@pcorpet @palfrey - you get to be my typing experts now.

**Comment by jamielennox** [comment]
released as 1.9.3 - thanks for the help, and sorry it took a little while.
